helm/reana-stable/values.yaml

```diff
@@ -0,0 +1,123 @@
+# Default values for reana.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# TODO Document this shuld be set for third party integrations and for
+# production deployments due to security issues.
+# reana_url: reana.cern.ch
+
+# Storage configuration
+# reana_url: reana.cern.ch
+debug:
+  enabled: false
+ui:
+  enabled: false
+eos:
+  enabled: false
+# CERN specific for now
+cephfs:
+  enabled: false
+  cephfs_volume_size: 200
+  access_modes: ReadWriteMany
+  provisioner: manila-provisioner
+  type: "Geneva CephFS Testing"
+  os_necret_name: os-trustee
+  os_necret_namespace: kube-system
+  cephfs_os_share_id: <cephfs-share-id>
+  cephfs_os_share_access_id: <cephfs-share-access-id>
+secrets:
+  database:
+    user: reana
+    # pasword: <my-db-pass>
+  gitlab:
+    REANA_GITLAB_OAUTH_APP_ID: reana
+    # REANA_GITLAB_OAUTH_APP_SECRET: <gitlab_oauth_app_secret>
+    REANA_GITLAB_HOST: gitlab.cern.ch
+  cern:
+    sso:
+      CERN_CONSUMER_KEY: reana
+      # CERN_CONSUMER_SECRET: <cern_consumer_secret
+
+# External database service configuration
+db_env_config:
+  REANA_DB_NAME: "reana"
+  REANA_DB_HOST: "reana-db"
+  REANA_DB_PORT: "5432"
+  # There are two more environment variables that should be set in order
+  # to connect to a database:
+  # REANA_DB_USERNAME: containing the database user name.
+  # REANA_DB_PASSWORD: containing the password for the user previously set.
+  # Both environment variables should be set inside a Kubernetes secret:
+  # `reana-db-secrets`
+
+# Filesystem related configuration variables
+volume_paths:
+  root_path: "/var/reana"
+  shared_volume_path: "/var/reana"
+  db_persistence_path: "/var/reana/db"
+
+# REANA components configuration
+components:
+  reana_server:
+    imagePullPolicy: IfNotPresent
+    image: reanahub/reana-server
+    tag: 0.6.0
```

Comment:
Hmm, perhaps we should not use 0.6.0 that was just released, if we rename components etc and if reana-cluster docs wouldn't work? Perhaps we should use something like 0.7.0.devYYYYMMDD or whatever dev version is compatible with SemVer 2 standard...

Reply:
(...sorry I mean that images can be 0.6.0 obviously, but the chart itself probably shouldn't get 0.6.0 label to avoid possible confusions...)

Reply:
Tested on a locally-compiled PROD-like instance after doing:

```console
$ sed -ie 's,0.6.0,latest,g' ./helm/reana-stable/values.yaml
```

Everything works :tada: Some observations:

(1) Is it necessary to include traefix tgz chart file in the git repo? Seeing:

```
Saving 1 charts
Downloading traefik from repo https://kubernetes-charts.storage.googleapis.com/ 
Deleting outdated charts
```

(2) Cosmetics: perhaps check for release presence to avoid a message like:

```
Error: uninstall: Release not loaded: reana: release: not found
```


(3) `helm/reana-stable` location of `helm` is nice, but I'm not sure about `reana-stable`, we can muse IRL more on this, e.g. about exposing minikube in the name and exposing cern in the name for customisations... e.g. easy way of amending values.yaml instead of my `sed` command above... e.g. allow passing values.yaml to the `make` target... These kinds of considerations. More IRL.

Reply:
Regarding semantic versioning, there are not development releases.

About changing the image versions to avoid this `sed` operation, I've tried two things to do this: firstly, I wanted to use templating inside `values.yaml` which is not possible for now (it will apparently be [in the near future](https://github.com/helm/helm/pull/6876)), the second was to use [`yaml` anchors](https://helm.sh/docs/chart_template_guide/yaml_techniques/#yaml-anchors), so there is a `global_tag` which is referenced in each component `tag` field, but this didn't work because when one modifies the `global_tag` with `helm install --set global_tag=0.7.0` the anchor is gone and it works just like a normal field (references stop working).

A possible solution could be to do the following (personally, not fully happy about it):
```yaml
...
# global_tag: 0.6.0
...
components:
  reana-server:
    image: reanahub/reana-server
    tag: 0.6.0
...
```
So the two are independent, then inside the templates (i.e. reana-server.yaml) we would give preference to `global_tag`, so if it is set, all images have the same version, if not we can have independent versions (from `.Values.components.[component_name].tag`)... it is not flexible nor intuitive but it would work. We could extract this discussion to a different issue to facilitate the merging.